### PR TITLE
Improvements getting a VCS' version

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/99)

### CODE REVIEW

File: downloader/src/funTest/kotlin/GitTest.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+package com.here.ort.downloader.vcs
+
+import com.here.ort.downloader.Expensive
+
+import io.kotlintest.matchers.shouldNotBe
+import io.kotlintest.specs.StringSpec
+
+class GitTest : StringSpec() {
+    init {
+        "Git version is detected correctly" {
```
Comment (mnonnenmacher) [inline downloader/src/funTest/kotlin/GitTest.kt]: Maybe rename to "Detected Git version is not empty" as this is what is tested?

File: downloader/src/funTest/kotlin/MercurialTest.kt [hunk 2]
```diff
@@ -44,6 +45,11 @@ class MercurialTest : StringSpec() {
     }
 
     init {
+        "Mercurial version is detected correctly" {
```
Comment (mnonnenmacher) [inline downloader/src/funTest/kotlin/MercurialTest.kt]: Maybe rename to "Detected Mercurial version is not empty" as this is what is tested?

File: downloader/src/funTest/kotlin/GitTest.kt (hunk 1 above)
Comment (sschuberth) [reply downloader/src/funTest/kotlin/GitTest.kt]: Done.

File: downloader/src/funTest/kotlin/MercurialTest.kt (hunk 2 above)
Comment (sschuberth) [reply downloader/src/funTest/kotlin/MercurialTest.kt]: Done.

File: downloader/src/funTest/kotlin/GitTest.kt (hunk 1 above)
Comment (sschuberth) [reply downloader/src/funTest/kotlin/GitTest.kt]: Done.

**Pull request merged**